pkg/compact/compact.go

```diff
@@ -1201,7 +1233,19 @@ func (c *BucketCompactor) Compact(ctx context.Context) (rerr error) {
 			go func() {
 				defer wg.Done()
 				for g := range groupChan {
-					shouldRerunGroup, _, err := g.Compact(workCtx, c.compactDir, c.planner, c.comp)
+					workCtx = tracing.CopyTraceContext(workCtx, ctx)
```

Comment:
I guess we don't need this. Just using the `workCtx` is fine and it has a tracer already.